### DISCUSSION TITLE
Ardurion 2.0 -> Compiling Failure

### DISCUSSION THREAD

**Discussion by a020088** [Zeppelin500/MBusino#13] [Q&A] [answered]
What libraries  you are using. If updating to the latest one I got compiling errors 
Especially the ArduionJson V7.0.2 fails. If I use 6.21.5 then it compiles at least.

What is the required environment?

**Comment by Zeppelin500** [comment]
Don't know Ardurion 2.0?
Arduino JSON switch to V7 last days and change the  API. With my Arduino IDE I have only compiler Warnings about this. It compiles free of failures. 
Next days I will switch to V7 MBusino and MBusinolib

**Comment by FieteRensen** [comment]
If you would like to use the V7 MBusino, which is available at https://github.com/Zeppelin500/MBusino/blob/main/playground/async_server/MBusinoAP_JSON7_asyncPortal_copy_pubsub/MBusinoAP_JSON7_asyncPortal_copy_pubsub.ino, you can use the following platformio.ini file to pull the current dependencies.
[platformio.ini.txt](https://github.com/Zeppelin500/MBusino/files/14073335/platformio.ini.txt)

**Comment by Zeppelin500** [comment]
MBusinoAP and MBusinoLib are now released with ArduinoJSON 7 support.

**Comment by a020088** [comment]
Still not compilable with ArduinoJson 6.21.5 and ArduinoJason7.0.2

ERROR 7.0.2
error: no matching function for call to 'ArduinoJson::V702PB2::JsonDocument::JsonDocument()'
   JsonDocument(JsonDocument&& src) : JsonDocument() {


ERROR V6.21.5
h:\Develop\Arduino\Sketchbook\libraries\ArduinoJson\src/ArduinoJson/Document/JsonDocument.hpp:296:3: error: 'ArduinoJson6150_0000010::JsonDocument::JsonDocument()' is protected
   JsonDocument() : _pool(0, 0) {
   ^
H:\Develop\Arduino\Sketchbook\MBusinoAP\MBusinoAP.ino:437:20: error: within this context
       JsonDocument jsonBuffer;
                    ^
H:\Develop\Arduino\Sketchbook\MBusinoAP\MBusinoAP.ino:438:50: error: no matching function for call to 'ArduinoJson6150_0000010::JsonDocument::add()'
       JsonArray root = jsonBuffer.add<JsonArray>();  

What Arduino 2.0 libraries are use using?

**Comment by Zeppelin500** [comment]
OK. All libraries the newest. 
What kind of hardware do you use?

**Comment by Zeppelin500** [comment]
Ah, the Lib is not distributed yet. Wait a couple of hours or load the two files in MBusinoLib/SRC/ manually to your Lib.

**Comment by Zeppelin500** [comment]
Dependency Graph
|-- MBusino @ 0.6.1+sha.0d248ee
|-- OneWire @ 2.3.7
|-- Adafruit Unified Sensor @ 1.1.14
|-- Adafruit BME280 Library @ 2.2.4
|-- ArduinoJson @ 7.0.2
|-- DallasTemperature @ 3.11.0
|-- TickTwo @ 4.4.0+sha.73b160b
|-- ESP Async WebServer @ 1.2.3+sha.f71e3d4
|-- PubSubClient @ 2.8.0+sha.2d228f2
|-- AsyncElegantOTA @ 2.2.8+sha.0d7e6f1
|-- ArduinoOTA @ 1.0
|-- DNSServer @ 1.1.1
|-- EEPROM @ 1.0
|-- ESP8266WiFi @ 1.0
|-- ESPAsyncTCP @ 1.2.2
|-- Wire @ 1.0

**Comment by FieteRensen** [comment, marked as answer]
You can flash the MBusino with https://web.esphome.io
1. Download the Bin from https://github.com/Zeppelin500/MBusino/blob/main/SRC/MBusinoAP/build/esp8266.esp8266.d1_mini/MBusinoAP.ino.bin
2. open https://web.esphome.io with the Chrome-Browser
3. Connect your 8266er with the COM-Port XX
4. Click on "INSTALL" not "PREPARE FOR FIRST USER
![Step4](https://github.com/Zeppelin500/MBusino/assets/32666091/7ff901e5-4994-49f9-a2c1-f914df4c7fb3)
5. Click on "Datei auswählen" and chose MBusinoAP.ino.bin
![Step5](https://github.com/Zeppelin500/MBusino/assets/32666091/221017c4-1a4b-4e81-8c4e-e0d9ce51a9d1)
6. Click on "INSTALL"
![Step6](https://github.com/Zeppelin500/MBusino/assets/32666091/b70db0ed-884b-4ca2-9f67-002f8dbabd9b)
7. Wait for finishing the Flash-Process
![Step7](https://github.com/Zeppelin500/MBusino/assets/32666091/5b327ce6-a832-4b6d-8381-7ecec4572459)
8. Flashing is completed successfully 
![Step8](https://github.com/Zeppelin500/MBusino/assets/32666091/782f9d0a-45da-4c94-b262-aa562115c390)
9. Search the MBusino Wlan Access-Point "MBusino Setup Portal" with your Smartphone
10. Configure your Mqtt-Broker and your WLAN

**Reply by Zeppelin500** [reply]
Thank you Fiete! Oder eben Danke. Sind, glaube ich, eh alle deutsch hier :-)

**Comment by a020088** [comment]
I think I will give up. It still will not compile. Too much tries w/o success.

**Comment by Zeppelin500** [comment]
Do you read the answers? There is a one minute solution without compiling.

**Comment by a020088** [comment]
I need to compile because I need to add some funcs for my application.

**Reply by Zeppelin500** [reply]
It compile without failures. I received today some feedback from users that there is no problem with compiling.

**Comment by a020088** [comment]
Final question: What Board are you using within Arduino?

**Comment by Zeppelin500** [comment]
from the Toturial, Inside the IDE:
**Kompilieren und flashen**
In der IDE unter dem Reiter "Werkzeuge", "Board" --> "Boardverwaltung" nach ESP8266 suchen und installieren. Danach unter Board das **Lolin(Wemos) D1 R2 & mini** auswählen.

in real: Anny d1 mini work fine
